lib/BulkDownloadCollection.php

```diff
@@ -0,0 +1,230 @@
+<?
+
+use Safe\DateTimeImmutable;
+
+use function Safe\preg_replace;
+
+/**
+ * @property string $LabelUrl
+ * @property string $UpdatedString
+ * @property array<BulkDownloadZipFile> $ZipFiles
+ */
+class BulkDownloadCollection{
+	use Traits\Accessor;
+
+	public Enums\BulkDownloadLabelType $LabelType;
+	public string $LabelName;
+	public string $LabelSort;
+	public ?string $LabelUrlSegment;
+	public int $EbookCount = 0;
+	public DateTimeImmutable $Updated;
+
+	protected ?string $_LabelUrl;
+	protected ?string $_UpdatedString;
+	/** @var array<BulkDownloadZipFile> $_ZipFiles */
+	protected array $_ZipFiles;
+
+	/** @var array<Ebook> $Ebooks */
+	public array $Ebooks;
+
+	protected function GetLabelUrl(): string{
+		if(isset($this->_LabelUrl)){
+			return $this->_LabelUrl;
+		}
+
+		switch($this->LabelType){
+			case Enums\BulkDownloadLabelType::Subject:
+				$this->_LabelUrl = '/subjects/' . $this->LabelUrlSegment;
+				break;
+			case Enums\BulkDownloadLabelType::Collection:
+				$this->_LabelUrl = '/collections/' . $this->LabelUrlSegment;
+				break;
+			case Enums\BulkDownloadLabelType::Author:
+				$this->_LabelUrl = '/ebooks/' . $this->LabelUrlSegment;
+				break;
+			case Enums\BulkDownloadLabelType::Month:
+				$this->_LabelUrl = '/months/' . $this->LabelUrlSegment;
+				break;
+		}
+
+		return $this->_LabelUrl;
+	}
+
+	protected function GetUpdatedString(): string{
+		if(isset($this->_UpdatedString)){
+			return $this->_UpdatedString;
+		}
+
+		$this->_UpdatedString = $this->Updated->format('M j');
+		// Add a period to the abbreviated month, but not if it's May (the only 3-letter month).
+		$this->_UpdatedString = preg_replace('/^(.+?)(?<!May) /', '\1. ', $this->_UpdatedString);
+		if($this->Updated->format('Y') != NOW->format('Y')){
+			$this->_UpdatedString = $this->Updated->format(Enums\DateTimeFormat::ShortDate->value);
+		}
+
+		return $this->_UpdatedString;
+	}
+
+	/**
+	 * @return array<BulkDownloadZipFile>
+	 */
+	protected function GetZipFiles(): array{
+		return $this->_ZipFiles ??= Db::Query('
+							SELECT *
+							from BulkDownloadZipFiles
+							where LabelType = ?
+								and LabelName = ?
+							order by Format
+					', [$this->LabelType, $this->LabelName], BulkDownloadZipFile::class);
+	}
+
+	public function AddZipFile(Enums\BulkDownloadFormatType $format, string $downloadUrl, int $downloadByteCount): void{
+		if(!isset($this->_ZipFiles)){
```

Comment:
You can just initialize this when declaring the property at the top of the class